test/e2e-secondary-network/secondary_network_test.go

```diff
@@ -223,6 +226,133 @@ func (data *testData) pingBetweenInterfaces(t *testing.T) error {
 	return nil
 }
 
+// getIPPoolNames retrieves the IPPool names associated with a given set of interfaces in a Pod. It queries the NetworkAttachmentDefinition for each interface 
+// and extracts the list of IPPools from its IPAM configurations. It returns a map where each interface name maps to its corresponding IPPools.
+func (data *testData) getIPPoolNames(interfaces []string, vlanPod *testPodInfo, namespace string) (map[string][]string, error) {
+	client, err := netattdef.NewForConfig(data.e2eTestData.KubeConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	ipPoolsMap := make(map[string][]string)
+
+	for _, iface := range interfaces {
+		networkName, exists := vlanPod.interfaceNetworks[iface]
+		if !exists {
+			return nil, fmt.Errorf("network name not found for interface %s", iface)
+		}
+		netAttachDef, err := client.K8sCniCncfIoV1().NetworkAttachmentDefinitions(namespace).Get(context.TODO(), networkName, metav1.GetOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("failed to get NetworkAttachmentDefinition %s for interface %s: %w", networkName, iface, err)
+		}
+
+		var netAttachDefConfig struct {
+			IPAM struct {
+				IPPools []string `json:"ippools"`
+			} `json:"ipam"`
+		}
+
+		if err := json.Unmarshal([]byte(netAttachDef.Spec.Config), &netAttachDefConfig); err != nil {
+			return nil, fmt.Errorf("failed to parse NetworkAttachmentDefinition config JSON for interface %s: %w", iface, err)
+		}
+
+		ipPoolsMap[iface] = netAttachDefConfig.IPAM.IPPools
+
+	}
+	return ipPoolsMap, nil
+}
+
+func (data *testData) checkIPReleased(ipPools map[string][]string, podIPsAndIfaces map[string]net.IP, interfaces []string) error {
+	crdClient := data.e2eTestData.CrdClient
+	
+	return wait.PollUntilContextTimeout(context.Background(), 5*time.Second, 40*time.Second, true, func(ctx context.Context) (bool, error) {
```

Comment:
I feel we should not wait for that long. At most 10 seconds, and m maybe poll every 1 seconds.